src/node_sqlite.cc

```diff
@@ -533,6 +726,116 @@ void DatabaseSync::Exec(const FunctionCallbackInfo<Value>& args) {
   CHECK_ERROR_OR_THROW(env->isolate(), db->connection_, r, SQLITE_OK, void());
 }
 
+// database.backup(destination, { sourceDb, targetDb, rate, progress: (total,
+// remaining) => {} )
+void DatabaseSync::Backup(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!args[0]->IsString()) {
+    THROW_ERR_INVALID_ARG_TYPE(
+        env->isolate(), "The \"destination\" argument must be a string.");
+    return;
+  }
+
+  int rate = 100;
+  std::string source_db = "main";
+  std::string dest_db = "main";
+
+  DatabaseSync* db;
+  ASSIGN_OR_RETURN_UNWRAP(&db, args.This());
+
+  THROW_AND_RETURN_ON_BAD_STATE(env, !db->IsOpen(), "database is not open");
+
+  Utf8Value destFilename(env->isolate(), args[0].As<String>());
+  Local<Function> progressFunc = Local<Function>();
+
+  if (args.Length() > 1) {
+    if (!args[1]->IsObject()) {
+      THROW_ERR_INVALID_ARG_TYPE(env->isolate(),
+                                 "The \"options\" argument must be an object.");
+      return;
+    }
+
+    Local<Object> options = args[1].As<Object>();
+    Local<String> progress_string =
+        FIXED_ONE_BYTE_STRING(env->isolate(), "progress");
+    Local<String> rate_string = FIXED_ONE_BYTE_STRING(env->isolate(), "rate");
+    Local<String> target_db_string =
+        FIXED_ONE_BYTE_STRING(env->isolate(), "targetDb");
+    Local<String> source_db_string =
+        FIXED_ONE_BYTE_STRING(env->isolate(), "sourceDb");
```

Comment:
Consider putting these on `env()` so you can do `env().targetdb_string()`, etc (see `env-properties.h`) .. that will avoid recreating these strings on every call.